modules/virt-about-application-consistent-backups.adoc

```diff
@@ -0,0 +1,13 @@
+// Module included in the following assemblies:
+//
+// * virt/backup_restore/virt-backup-restore-snapshots.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="virt-about-application-consistent-backups_{context}"]
+= About application-consistent snapshots and backups
+
+You can configure application-consistent snapshots and backups for Linux or Windows virtual machines (VMs) through a cycle of freezing and thawing. For any application, you can either configure a script on a Linux VM or register on a Windows VM to be notified when a snapshot or backup is due to begin.
+
+On a Linux VM, freeze and thaw processes trigger automatically when a snapshot is taken or a backup is started by using, for example, a plugin from Velero or another backup vendor. The freeze process, performed by QEMU Guest Agent (QEMU GA) freeze hooks, ensures that before the snapshot or backup of a VM occurs, all of the VM's filesystems are frozen and each appropriately configured application is informed that a snapshot or backup is about to start. This notification affords each application the opportunity to quiesce its state. Depending on the application, quiescing might involve temporarily refusing new requests, finishing in-progress operations, and flushing data to disk. The operating system is then directed to quiesce the filesystems by flushing outstanding writes to disk and freezing new write activity. All new connection requests are refused. When all applications have become inactive, the QEMU GA freezes the filesystems, and a snapshot is taken or a backup initiated. After the taking of the snapshot or start of the backup, the thawing process begins. Filesystems writing is reactivated and applications receive notification to resume normal operations.
+
+The same cycle of freezing and thawing is available on a Windows VM. Applications register with the Volume Shadow Copy Service (VSS) to receive notifications that they should flush out their data because a backup or snapshot is imminent. Thawing of the applications after the backup or snapshot is complete returns them to an active state.
```

Comment:
Should we include a link to Microsoft documentation (ie. https://learn.microsoft.com/en-us/windows-server/storage/file-server/volume-shadow-copy-service)?

Reply:
Thanks, @aglitke. I'll send this PR along to QE.

We do try to avoid deep external links that might break in the future, so I'm going to suggest something like, "See the Windows Server documentation about the Volume Shadow Copy Service" for more details."